### PR TITLE
Custom protocol handlers for fetch

### PR DESCRIPTION
**Pull request by modulesio**
Continuation of https://github.com/exokitxr/exokit/pull/1250 with cleanups.

### CODE REVIEW

File: src/Worker.js [hunk 1]
```diff
@@ -10,11 +10,13 @@ const {
 } = require('worker_threads');
 
 const {createImageBitmap} = require('./DOM.js');
-const fetch = require('window-fetch');
+const {fetch} = require('./fetch');
 const {XMLHttpRequest} = require('window-xhr');
 const WebSocket = require('ws/lib/websocket');
 const {FileReader} = require('./File.js');
 
+XMLHttpRequest.setFetchImplementation(fetch)
```
Comment (modulesio) [inline src/Worker.js]: I think we don't need this copy. `Worker` inherits from `WindowBase`.

File: src/WindowBase.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ const {CustomEvent, DragEvent, ErrorEvent, Event, EventTarget, KeyboardEvent, Me
 const {MediaDevices, Clipboard, Navigator} = require('./Navigator');
 const {FileReader} = require('./File');
 const {XMLHttpRequest, FormData} = require('window-xhr');
-const fetch = require('window-fetch');
+const {fetch, setBaseFetchUrl} = require('./fetch');
```
Comment (modulesio) [inline src/WindowBase.js]: Dead `setBaseFetchUrl`.

**Comment by modulesio** [comment]
This is failing CI:

```
  1) ScriptElement
       "before each" hook for "HTML tags":
     TypeError: URL.lookupObjectURL is not a function
      at fetch (/app/src/fetch.js:17:22)
      at _fetchTextFollow (/app/src/core.js:35:35)
      at Object.exokit.load (/app/src/core.js:61:10)
      at Context.<anonymous> (/app/tests/unit/dom/ScriptElement.test.js:8:27)
      at callFn (/app/node_modules/mocha/lib/runnable.js:387:21)
      at Hook.Runnable.run (/app/node_modules/mocha/lib/runnable.js:379:7)
      at next (/app/node_modules/mocha/lib/runner.js:384:10)
      at Immediate.<anonymous> (/app/node_modules/mocha/lib/runner.js:425:5)
      at processImmediate (internal/timers.js:439:21)
      at process.topLevelDomainCallback (domain.js:126:23)

  2) ScriptElement
       "after each" hook for "HTML tags":
     TypeError: Cannot read property 'destroy' of undefined
      at Context.<anonymous> (/app/tests/unit/dom/ScriptElement.test.js:25:19)
      at callFn (/app/node_modules/mocha/lib/runnable.js:387:21)
      at Hook.Runnable.run (/app/node_modules/mocha/lib/runnable.js:379:7)
      at next (/app/node_modules/mocha/lib/runner.js:384:10)
      at Immediate.<anonymous> (/app/node_modules/mocha/lib/runner.js:425:5)
      at processImmediate (internal/timers.js:439:21)
      at process.topLevelDomainCallback (domain.js:126:23)
```

**Comment by RangerMauve** [comment]
Oh dang.

**Comment by modulesio** [comment]
@RangerMauve No need, should be fixed now. Thanks!

**Pull request merged**